src/sql/query.rs

```diff
@@ -3541,8 +3582,10 @@ pub fn scalar_type_from_sql(data_type: &DataType) -> Result<ScalarType, failure:
         DataType::Interval => ScalarType::Interval,
         DataType::Bytea => ScalarType::Bytes,
         DataType::Jsonb => ScalarType::Jsonb,
-        other @ DataType::Array(_)
-        | other @ DataType::Binary(..)
+        DataType::List(elem_type) => ScalarType::List(Box::new(
+            ColumnType::new(scalar_type_from_sql(elem_type)?).nullable(true),
```

Comment:
Why always nullable? Are those just the semantics?

Reply:
Yeah, it's copied from arrays. I think it's just because there is no way in the postgresql syntax to construct a nonnullable/nullable type except in a table column.

Reply:
Like it would have to be `LIST[1] :: INT NULLABLE LIST` or something.

Reply:
What's the difference between a `ColumnType` and `ScalarType`? Should the parameter not be the second? Types don't really have a notion of nullability AIUI

Reply:
Just checked the docs for sql records and those also don't allow `NOT NULL`.

Reply:
Wait, yeah, I think I agree with Justin that it might be more natural for these to embed the `ScalarType` directly, rather than the ColumnType?

Reply:
Ok. I wavered between the two to begin with, happy to change it.